flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -783,9 +816,17 @@ public void configure(ReadableConfig configuration) {
         configuration
                 .getOptional(ExecutionCheckpointingOptions.CHECKPOINT_ID_OF_IGNORED_IN_FLIGHT_DATA)
                 .ifPresent(this::setCheckpointIdOfIgnoredInFlightData);
-        configuration
-                .getOptional(ExecutionCheckpointingOptions.ALIGNMENT_TIMEOUT)
-                .ifPresent(this::setAlignmentTimeout);
+
+        Optional<Duration> alignedCheckpointTimeout =
+                configuration.getOptional(ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT);
+        if (alignedCheckpointTimeout.isPresent()) {
```

Comment:
IIRC this whole block is unnecessary. It is already implemented inside of the Configuration if you use the `deprecatedKeys`.

It should be enough to do:
```
Duration alignedCheckpointTimeout = configuration.get(ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT);
```